### ISSUE TITLE
Create a BLENDER_materials_nodes extension

### ISSUE THREAD

**Issue by Moguri** [Kupoman/blendergltf#82]
This extension would export shader nodes to be consumed by an engine. This is easier to consume than raw GLSL, but it still allows for shader creation in Blender.

**Comment by reduz** [comment]
sounds great, with Cycles and Eevee support please! :D

**Comment by stevenvergenz** [comment]
Easier to consume by which engines? Only Blender would be able to make use of these, would probably be easier to stick that data in extras than formalizing an extension.

**Comment by reduz** [comment]
@stevenvergenz Any engines. In Godot this would work great, in fact. There isn't really any kind of abstract shader language that would work, and writing directly to something like GLSL or SPIR-V is useless due to them being too low level.

If the node graph is exported, however, it is easy for a parser to convert it to the shaders used internally by any engine.

**Comment by Moguri** [comment]
I have created a material_nodes branch to begin work on the spec for this extension. No implementation work has started yet.

@reduz My current thought is to export a graph with each graph node having a type and a list of inputs and outputs (i.e., sockets). These sockets also have a type and value. The types on the sockets should be fairly well defined (e.g., color, scalar, node reference), but interpreting the node type will be mostly up to the consuming application. Does this rough cut align with what you were expecting? I have also listed you as a contributor for this extension.

**Comment by reduz** [comment]
Yeah.. a graph would work best as I can convert it to either godot shader
language or visual shader (if i succeed in implementing the same..)

**Comment by ghost** [comment]
I'm very sorry if my question doesn't fit the topic, but since I'm kind of new to github and programming stuff - maybe you can help me out.

I came from the Khronos glTF-Exporter for Blender to see, if the animation exportation of this exporter works properly - and it does. Very intuitive and reliable, so thanks for that!

My problem is, that with the Khronos exporter and their KHR-Shader Groups I could easily create materials that looked a lot like in Blender-Cycles, when rendered via three.js (WebGL). 
Until now, I couldn't figure out a way to create materials with your exporter - all objects are displayed shiny (100% metallic, 0% roughness, color: ffffff). Do you have a documentation or something like that somewhere? I wasn't able to find it yet.

Thanks and best regards

**Comment by Kupoman** [comment]
@dustronauts This exporter exports Blender Internal materials, but not Cycles materials. If you look at the material panel while the renderer is set to "Blender Render", you should see a "PBR Export" panel that allows you to set some PBR related settings.
